src/main/java/rua/task/TaskList.java

```diff
@@ -0,0 +1,140 @@
+package rua.task;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+
+public class TaskList {
+    private final ArrayList<Task> tasks;
+
+    /**
+     * Constructs a TaskList object (with no tasks inside by default).
+     */
+    public TaskList() {
+        this.tasks = new ArrayList<>();
+    }
+
+    /**
+     * Constructs a Task object.
+     *
+     * @param tasks An ArrayList of Task to represent the existing tasks in the TaskList or
+     *              tasks that will be added into the TaskList.
+     */
+    public TaskList(ArrayList<Task> tasks) {
+        this.tasks = tasks;
+    }
+
+    /**
+     * Returns the ArrayList of Task objects in the TaskList.
+     *
+     * @return The ArrayList of Task objects in the TaskList.
+     */
+    public ArrayList<Task> getTasks() {
+        return this.tasks;
+    }
+
+    /**
+     * Returns a new TaskList object with updated list of tasks.
+     *
+     * @param tasks An ArrayList of Task objects in the new TaskList.
+     * @return The new TaskList object with updated list of tasks.
+     */
+    private TaskList update(ArrayList<Task> tasks) {
+        return new TaskList(tasks);
+    }
+
+    /**
+     * Adds a Task object to the TaskList.
+     *
+     * @param task A Task object that will be added into the TaskList.
+     * @return The new TaskList object with the new Task.
+     */
+    public TaskList add(Task task) {
```

Comment:
Perhaps can make the method name more specific
```suggestion
    public TaskList addTask(Task task) {
```